### PR TITLE
Extensions integrations testing

### PR DESCRIPTION
**Pull request by theopolis**
1. Removes sdk as a specialized build output, libosquery.a is the output of `make sdk` and an intermediate required target for `make`. They build the same library, with the same defines.
2. To accomplish this "one-libosquery.a-to-rule-them-all" the SQL implementation is a registry type/item.
3. Extensions unittests.

### CODE REVIEW

**Comment by zwass** [comment]
Is `test_extensions.py` intended to be run manually after a build of the source? Can we have it run as part of `make test`?

**Comment by theopolis** [comment]
Manually right now, it's mostly a tool to help develop the extensions workflow assumptions. I'd love a more robust way to discover and run various integrations. That python script does some argparsing and thrift importing that might be hard to run automatically since it depends on a variable /build/platform/generated/gen-py path. If we hard-assume that path then a unittest discovery would work fine. :)

**Comment by marpaia** [comment]
Should we remove `SDK=True` here: https://github.com/facebook/osquery/blob/master/tools/sync.sh#L41

File: osquery/core/tables.cpp [hunk 1]
```diff
@@ -197,5 +135,88 @@ std::string TablePlugin::statement() const {
...
+    boost::property_tree::ptree child;
+    child.put("op", constraint.op);
+    child.put("expr", constraint.expr);
+    expressions.push_back(std::make_pair("", child));
+  }
+  tree.add_child("list", expressions);
+  tree.put("affinity", affinity);
+}
+
+void ConstraintList::unserialize(const boost::property_tree::ptree& tree) {
```
Comment (marpaia) [inline osquery/core/tables.cpp]: `deserialize`?

File: osquery/core/tables.cpp [hunk 2]
```diff
@@ -197,5 +135,88 @@ std::string TablePlugin::statement() const {
...
+    }
+    if (!aggregate) {
+      // Speed up comparison.
+      return false;
+    }
+  }
+  return true;
+}
+
+std::set<std::string> ConstraintList::getAll(ConstraintOperator op) const {
```
Comment (marpaia) [inline osquery/core/tables.cpp]: `ConstraintOperator& op`?

**Comment by marpaia** [comment]
http://i.imgur.com/702YRTB.gif

File: osquery/core/tables.cpp (hunk 2 above)
Comment (theopolis) [inline osquery/core/tables.cpp]: hmm, the operator is an enum, not sure if passing the reference saves much :/ int might be smaller, ha!

**Comment by theopolis** [comment]
18wheel-youdidit

**Comment by theopolis** [comment]
No to the comment above about removing `SDK=True`, the `make sdk` is still a thing, it just limits the number of compile targets :icecream:

File: tools/codegen/gentargets.py [hunk 3]
```diff
@@ -16,12 +16,16 @@ def get_files_to_compile(json_data):
     for element in json_data:
         filename = element["file"]
         if not filename.endswith("_tests.cpp") and \
-            "third-party" not in filename and \
-            "generated" not in filename:
-            filename = filename.replace(REPO_ROOT_DIR + "/osquery/", "")
-            filename = filename.replace(REPO_ROOT_DIR + "/build/sync/code-analysis/", "")
+                "third-party" not in filename and \
+                "example" not in filename and \
+                "generated/gen" not in filename:
```
Comment (theopolis) [inline tools/codegen/gentargets.py]: this grabs the utils_amalgamated.cpp file, needed for basic utility tables (which are not broadcasted from core->extensions).

**Pull request merged**